util-core/src/main/scala/com/twitter/util/Event.scala

```diff
@@ -304,20 +304,27 @@ trait Event[+T] { self =>
   }
 
   /**
-   * Progressively build a collection of events using the passed-in
+   * Progressively build any collection of events using the passed-in
    * builder. A value containing the current version of the collection
    * is notified for each incoming event.
    */
-  def build[U >: T, That](implicit cbf: CanBuild[U, That]): Event[That] = new Event[That] {
+  def buildAny[That](implicit factory: Factory[T, That]): Event[That] = new Event[That] {
     def register(s: Witness[That]): Closable = {
-      val b = cbf()
+      val b = factory.newBuilder
```

Comment:
The contract of builder makes calling `result()` more than once undefined behaviour (with the exception of `ReusableBuilder` which can call `result()` again after calling `clear`). That means that this relies upon (and always has relied upon) builder being well-behaved in cases where it doesn't have to play that nice according to its contract.